### PR TITLE
REVSDL-2575 RRDL returns notification instead of response to mobile

### PR DESCRIPTION
**Pull request by thinhts**
@KKarlash 
@giangtqh 
@DKlimenko 
@bonewell 
@VladSemenyuk

### CODE REVIEW

File: src/components/vr_cooperation/src/command/register_service_request.cc [hunk 1]
```diff
@@ -75,9 +75,14 @@ void RegisterServiceRequest::ProcessEvent(
 
 void RegisterServiceRequest::SendNotificationToHMI() {
   LOG4CXX_AUTO_TRACE(logger_);
-  application_manager::MessagePtr json_msg = json_message();
-  json_msg->set_message_type(application_manager::MessageType::kNotification);
-  commands::Command* command = CommandFactory::Create(parent(), json_msg);
+  application_manager::MessagePtr request_json_msg = json_message();
+  application_manager::Message* notification_message =
+      new application_manager::Message(*(request_json_msg.get()));
```
Comment (bonewell) [inline src/components/vr_cooperation/src/command/register_service_request.cc]: SharedPtr has operator *

**Comment by bonewell** [comment]
It's strange. Why don't you send back response?

**Comment by thinhts** [comment]
@bonewell I'll describe this bug:
Cause: Because we need to send response to mobile and notification to hmi.
First we use this message with response type to send to mobile. But it will wait in message loop.
Second we also use above message and change message type to notification to send to hmi.
After that message loop thread above send this message which has already changed message type to mobile. So at mobile side, It receive notification instead of response
Solution: Create a new message for notification.

**Review by bonewell — APPROVED** [review]

**Review by bonewell — APPROVED** [review]

**Review by bonewell — APPROVED** [review]

**Comment by KKarlash** [comment]
Reviewed

**Comment by bonewell** [comment]
Reviewed.

**Pull request merged**